apis/datastream/v1alpha1/privateconnection_types.go

```diff
@@ -0,0 +1,166 @@
+// +kubebuilder:printcolumn:name="Ready",JSONPath=".status.conditions[?(@.type=='Ready')].status",type="string",description="When 'True', the most recent reconcile of the resource succeeded"
+// +kubebuilder:printcolumn:name="Status",JSONPath=".status.conditions[?(@.type=='Ready')].reason",type="string",description="The reason for the value in 'Ready'"
+// +kubebuilder:printcolumn:name="Status Age",JSONPath=".status.conditions[?(@.type=='Ready')].lastTransitionTime",type="date",description="The last transition time for the value in 'Status'"
+
+// DatastreamPrivateConnection is the Schema for the DatastreamPrivateConnection API
+// +k8s:openapi-gen=true
+type DatastreamPrivateConnection struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec   DatastreamPrivateConnectionSpec   `json:"spec,omitempty"`
+	Status DatastreamPrivateConnectionStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// DatastreamPrivateConnectionList contains a list of DatastreamPrivateConnection
+type DatastreamPrivateConnectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DatastreamPrivateConnection `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&DatastreamPrivateConnection{}, &DatastreamPrivateConnectionList{})
+}
+
+// +kcc:proto=google.cloud.datastream.v1.Error
+type Error struct {
+	// A title that explains the reason for the error.
+	// +kcc:proto:field=google.cloud.datastream.v1.Error.reason
+	Reason *string `json:"reason,omitempty"`
+
+	// A unique identifier for this specific error,
+	//  allowing it to be traced throughout the system in logs and API responses.
+	// +kcc:proto:field=google.cloud.datastream.v1.Error.error_uuid
+	ErrorUUID *string `json:"errorUUID,omitempty"`
+
+	// A message containing more information about the error that occurred.
+	// +kcc:proto:field=google.cloud.datastream.v1.Error.message
+	Message *string `json:"message,omitempty"`
+
+	// The time when the error occurred.
+	// +kcc:proto:field=google.cloud.datastream.v1.Error.error_time
+	ErrorTime *string `json:"errorTime,omitempty"`
+
+	// Additional information about the error.
+	// +kcc:proto:field=google.cloud.datastream.v1.Error.details
+	Details map[string]string `json:"details,omitempty"`
+}
+
+// +kcc:proto=google.cloud.datastream.v1.VpcPeeringConfig
+type VpcPeeringConfig struct {
+	// Required. Fully qualified name of the VPC that Datastream will peer to.
+	//  Format: `projects/{project}/global/{networks}/{name}`
+	// +kcc:proto:field=google.cloud.datastream.v1.VpcPeeringConfig.vpc
+	NetworkRef *refsv1beta1.ComputeNetworkRef `json:"networkRef,omitempty"`
+
+	// Required. A free subnet for peering. (CIDR of /29)
+	// +kcc:proto:field=google.cloud.datastream.v1.VpcPeeringConfig.subnet
+	Subnet *string `json:"subnet,omitempty"`
```

Comment:
Could you check if this should be a ref to https://cloud.google.com/config-connector/docs/reference/resource-docs/compute/computesubnetwork

Reply:
As far as I can tell, `Subnet` is not a reference field.

```
vpcPeeringConfig:
    networkRef:
        xxx
    subnet: "10.0.0.0/29"
```

Reply:
According to [1]:

> Identify an available IP range (with a CIDR block of /29) on the VPC network. This can't be an IP range that already exists as a subnet, a private services access pre-allocated IP range, or any route (other than the default 0.0.0.0 route) that includes the IP range. Datastream uses this IP range to create a subnet so that it can communicate with the source database. The following table describes valid IP ranges.

[1] https://cloud.google.com/datastream/docs/create-a-private-connectivity-configuration#before-you-begin